app/static/src/app/components/fit/support.ts

```diff
@@ -1,31 +1,44 @@
-import { ModelFitRequirements } from "../../store/modelFit/state";
+import { ModelFitRequirements, FitUpdateRequiredReasons } from "../../store/modelFit/state";
 import userMessages from "../../userMessages";
 import { joinStringsSentence } from "../../utils";
 
+const appendIf = (container: string[], test: boolean, value: string) => {
+    if (test) {
+        container.push(value);
+    }
+};
+
 export const fitRequirementsExplanation = (reqs: ModelFitRequirements): string => {
-    const reasons: string[] = [];
-    const appendIf = (test: boolean, value: string) => {
-        if (test) {
-            reasons.push(value);
-        }
-    };
+    const explanation: string[] = [];
     const help = userMessages.modelFit.fitRequirements;
-    appendIf(!reqs.hasModel, help.needsModel);
-    appendIf(!reqs.hasData, help.needsData);
+    appendIf(explanation, !reqs.hasModel, help.needsModel);
+    appendIf(explanation, !reqs.hasData, help.needsData);
     // Can only tell the user to set a time variable if we have data
-    appendIf(reqs.hasData && !reqs.hasTimeVariable, help.needsTimeVariable);
+    appendIf(explanation, reqs.hasData && !reqs.hasTimeVariable, help.needsTimeVariable);
     // Can only tell the user to link variables if they have model and data
-    appendIf(reqs.hasModel && reqs.hasData && !reqs.hasLinkedVariables, help.needsLinkedVariables);
+    appendIf(explanation, reqs.hasModel && reqs.hasData && !reqs.hasLinkedVariables, help.needsLinkedVariables);
     // Can only tell the user to set a target if we have all of the
     // above (checking hasLinkedVariables is sufficient)
-    appendIf(reqs.hasLinkedVariables && !reqs.hasTarget, help.needsTarget);
+    appendIf(explanation, reqs.hasLinkedVariables && !reqs.hasTarget, help.needsTarget);
     // Can only tell the user to change parameters if we have a
     // model. This shold be announced last as it's on the tab they are
     // looking at
-    appendIf(reqs.hasModel && !reqs.hasParamsToVary, help.needsParamsToVary);
+    appendIf(explanation, reqs.hasModel && !reqs.hasParamsToVary, help.needsParamsToVary);
 
     // Fallback reason if something unexpected has happened.
-    appendIf(reasons.length === 0, help.unknown);
+    appendIf(explanation, explanation.length === 0, help.unknown);
 
-    return `${help.prefix} ${joinStringsSentence(reasons)}.`;
+    return `${help.prefix} ${joinStringsSentence(explanation)}.`;
+};
+
+export const fitUpdateRequiredExplanation = (reasons: FitUpdateRequiredReasons): string => {
+    const explanation: string[] = [];
+    const help = userMessages.modelFit.updateFitReasons;
+    appendIf(explanation, reasons.modelChanged, help.modelChanged);
+    appendIf(explanation, reasons.dataChanged, help.dataChanged);
+    appendIf(explanation, reasons.linkChanged && !reasons.modelChanged && !reasons.dataChanged, help.linkChanged);
+    appendIf(explanation, reasons.parameterChanged && !reasons.modelChanged, help.parameterChanged);
```

Comment:
Yes, this makes sense, I think, because if they've changed the model or data then the link and parameter values are potentially invalidated anyway. I suppose there's an argument for including everything that's changed so that the user can see consistently what things cause invalidation... but it's also preferable not to overwhelm with long list of reasons..